src/core/pointcloud/qgseptdecoder.cpp

```diff
@@ -672,4 +672,303 @@ QgsPointCloudBlock *QgsEptDecoder::decompressLaz( const QByteArray &byteArrayDat
+        case LazAttribute::Green:
+          _storeToStream<unsigned short>( dataBuffer, outputOffset, requestedAttribute.type, rgb.g );
+          break;
+        case LazAttribute::Blue:
+          _storeToStream<unsigned short>( dataBuffer, outputOffset, requestedAttribute.type, rgb.b );
+          break;
+        case LazAttribute::ExtraBytes:
+        {
+          switch ( requestedAttribute.type )
+          {
+            case QgsPointCloudAttribute::Char:
+              _storeToStream<char>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<char * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::UChar:
+              _storeToStream<unsigned char>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<unsigned char * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::Short:
+              _storeToStream<qint16>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<qint16 * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::UShort:
+              _storeToStream<quint16>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<quint16 * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::Int32:
+              _storeToStream<qint32>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<qint32 * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::UInt32:
+              _storeToStream<quint32>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<quint32 * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::Int64:
+              _storeToStream<qint64>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<qint64 * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::UInt64:
+              _storeToStream<quint64>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<quint64 * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::Float:
+              _storeToStream<float>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<float * >( &buf[requestedAttribute.offset] ) );
+              break;
+            case QgsPointCloudAttribute::Double:
+              _storeToStream<double>( dataBuffer, outputOffset, requestedAttribute.type, *reinterpret_cast<double * >( &buf[requestedAttribute.offset] ) );
+              break;
+          }
+        }
+        break;
+        case LazAttribute::MissingOrUnknown:
+          // just store 0 for unknown/missing attributes
+          _storeToStream<unsigned short>( dataBuffer, outputOffset, requestedAttribute.type, 0 );
+          break;
+      }
```

Comment:
Also a lot of duplication here - most of the internal part of the for-loop could go to a separate function to unpack and store data to stream...